_data-prepper/pipelines/configuration/processors/mutate-strings.md

```diff
@@ -0,0 +1,237 @@
+
+## uppercase_string
+
+The `uppercase_string` processor converts a key to uppercase.
+
+<!--- Converts a key or a string? Are only keys and strings being converted?--->
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - uppercase_string:
+        with_keys:
+          - "uppercaseField"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the correct file path:
+
+```json
+{"uppercaseField": "hello"}
+```
+When you run Data Prepper with this `pipeline.yaml` file, you should see the following output:
+
+```json
+{"uppercaseField": "HELLO"}
+```
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
+
+* `with_keys` (required): A list of keys to convert to uppercase.
+
+## Lowercase string
+
+The `lowercase string` processor converts a string to lowercase.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - lowercase_string:
+        with_keys:
+          - "lowercaseField"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"lowercaseField": "TESTmeSSage"}
+```
+When you run Data Prepper with this `pipeline.yaml` file, you will see the following output:
+
+```json
+{"lowercaseField": "testmessage"}
+```
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
+
+* `with_keys` (required): A list of keys to convert to lowercase
+
+## trim_string
+
+The `trim_string` processor removes whitespace from the beginning and end of a key.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - trim_string:
+        with_keys:
+          - "trimField"
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the your file path:
+
+```json
+{"trimField": " Space Ship "}
+```
+When you run Data Prepper with this `pipeline.yaml` file, you will see the following output:
+
+```json
+{"trimField": "Space Ship"}
+```
+
+### Configuration
+
+<!--- Are these paramters? How are these items used? Need some intro text here.--->
```

Comment:
Same as previous Configuration sections. What does configuration specifically allow the user to do here?